### ISSUE TITLE
Lint/LiteralAsCondition false positive in case...when

### ISSUE THREAD

**Issue by jcoyne** [rubocop/rubocop#14727]
Currently it is tripping on this code:

```
lib/blacklight.rb:36:48: W: Lint/LiteralAsCondition: Literal 'solr' appeared as a condition.
    case connection_config&.fetch(:adapter) || 'solr'
                                               ^^^^^^
```

I don't think a `case` should apply here as this is more akin to a default return value from a function.  Whatever is present in this clause is compared to the `when` clauses after this.

Using rubocop-1.77.0

**Comment by jcoyne** [comment]
No longer an issue in 1.82.0
